src/translate_c.zig

```diff
@@ -5942,6 +5943,25 @@ fn parseCPrimaryExprInner(c: *Context, m: *MacroCtx, scope: *Scope) ParseError!N
 
 fn parseCPrimaryExpr(c: *Context, m: *MacroCtx, scope: *Scope) ParseError!Node {
     var node = try parseCPrimaryExprInner(c, m, scope);
+    while (node.castTag(.identifier)) |identifier_tag| {
+        // If the given macro is blank, simply ignore it and skip to the next.
+        if (c.global_scope.blank_macros.contains(identifier_tag.data)) {
+            switch (m.peek().?) {
+                .Nl, .Eof => {
+                    // Every token in this macro is blank,
+                    // so we can treat this macro as also blank.
```

Comment:
This doesn't seem correct since we already had to parse a primary expression to get here.

Reply:
Yes, I think that comment is probably worded incorrectly. That line isn't reached for blank macros, it is reached for macros where every token is an identifier, and all of those identifiers themselves identify blank macros. Example:
```
#define blank
#define blank_ blank blank blank
```
I will clarify the comment.

Reply:
I've reworked this, see below.